Exercise_2.py

```diff
@@ -1,23 +1,38 @@
-# Python program for implementation of Quicksort Sort 
+# Python program for implementation of Quicksort Sort
   
 # give you explanation for the approach
+#In this program we set up the last elemnet in the array as pivot.
+#We place the pivot in its current position and check if the element is smaller or larger then it.
+# If it is smaller it is placed towards left og the pivot else right.
 def partition(arr,low,high):
+    i=(low-1)# index of the smaller element
+    pivot=arr[high] #setting pivot
```

Comment:
What should be the pivot if we want to have the most efficient version of quicksort ?

Reply:
For the most efficient version of quicksort, pivot should be chosen as the middle element in the array, such that the partition is evenly balanced. Because, here we evaluate on basis of time for partitioning. Correct me if I am wrong.